Tests/TuistDependenciesTests/SwiftPackageManager/Utils/PackageInfoMapperTests.swift

```diff
@@ -241,20 +248,144 @@ final class PackageInfoMapperTests: TuistUnitTestCase {
                 targets: [
                     .test(
                         "Target1",
-                        headers: .init(public: "/Package/Path/Sources/Target1/include/Public.h")
+                        customSettings: [
+                            "HEADER_SEARCH_PATHS": ["$(SRCROOT)/Sources/Target1/include"],
+                        ],
+                        moduleMap: "/Package/Path/Sources/Target1/include/module.modulemap"
+                    ),
+                ]
+            )
+        )
+    }
+
+    func testMap_whenHasHeadersWithUmbrellaHeader() throws {
+        fileHandler.stubFilesAndDirectoriesContained = { path in
```

Comment:
I recommend doing IO in tests like this one. It turns the unit test into an integration test because we are hitting the system, and that slows the test a bit, but it'll make the test easier to maintain and closer to a real scenario. Here's the approach that I'm proposing and that we do in other places:

1. Get the temporary directory. The `TuistUnitTestCase` has a method for that.
2. Create the files scenario (e.g there's a headers directory in the temporary directory with these headers)
3. Run the subject pointing to that temporary directory
4. Assert that the module map has been created at the right path and with the right content

Note that the temporary directory will get automatically cleaned after the test is done.

Reply:
Done ✅